Loop/View Controllers/SettingsTableViewController.swift

```diff
@@ -438,21 +437,20 @@ final class SettingsTableViewController: UITableViewController {
             let row = ConfigurationRow(rawValue: indexPath.row)!
             switch row {
             case .carbRatio:
-                let scheduleVC = DailyQuantityScheduleTableViewController()
-
-                scheduleVC.delegate = self
-                scheduleVC.title = NSLocalizedString("Carb Ratios", comment: "The title of the carb ratios schedule screen")
-                scheduleVC.unit = .gram()
-
-                if let schedule = dataManager.loopManager.carbRatioSchedule {
-                    scheduleVC.timeZone = schedule.timeZone
-                    scheduleVC.scheduleItems = schedule.items
-                    scheduleVC.unit = schedule.unit
-                } else if let timeZone = dataManager.pumpManager?.status.timeZone {
-                    scheduleVC.timeZone = timeZone
-                }
+                let editor = CarbRatioScheduleEditor(
+                    schedule: dataManager.loopManager.carbRatioSchedule,
+                    onSave: { [dataManager] newSchedule in
+                        dataManager?.loopManager.carbRatioSchedule = newSchedule
+                        dataManager?.analyticsServicesManager.didChangeCarbRatioSchedule()
+                        tableView.reloadRows(at: [indexPath], with: .automatic)
+                    }
+                )
+
+                let hostingController = DismissibleHostingController(rootView: editor, onDisappear: {
+                    tableView.deselectRow(at: indexPath, animated: true)
+                })
```

Comment:
When I saw this before I didn't comment on it. But now I'm curious. Why deselect the row when after the `editor` view disappears? Or am I reading this wrong?

Reply:
Deselecting the row is necessary to prevent the row from appearing 'pressed' after the modal is dismissed.